helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -255,11 +261,14 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
       Map<HelixConstants.ChangeType, Set<String>> clusterChanges, Map<String, Resource> resourceMap,
       final CurrentStateOutput currentStateOutput) throws HelixRebalanceException {
     LOG.info("Start calculating the new best possible assignment.");
+    LatencyMetric partialRebalanceMetric = new PartialRebalanceMetric("PartialRebalanceLatency");
+    _metricCollector.addMetric(partialRebalanceMetric);
```

Comment:
I would prefer adding metrics in the constructor and use getMetric() here. The code would be easier to understand.
Or if you prefer, we can just merge get and add methods.

The problem of the implementation is that, if the rebalancer calls partialRebalance more than once, the update metric object will not be recorded in the Metric Collector.